Test/Utils/ComponentsConfiguration.ps1

```diff
@@ -0,0 +1,126 @@
+. $PSScriptRoot\..\..\Common\Init.ps1
+. $PSScriptRoot\..\..\Common\Invoke-CommandWithFunctions.ps1
+. $PSScriptRoot\..\CIScripts\Testenv\Testenv.ps1
+
+function Get-DefaultCNMPluginsConfigPath {
+    return "C:\ProgramData\Contrail\etc\contrail\cnm-driver.conf"
+}
+
+function New-CNMPluginConfigFile {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session,
+        [Parameter(Mandatory = $true)] [string] $AdapterName,
+        [Parameter(Mandatory = $true)] [OpenStackConfig] $OpenStackConfig,
+        [Parameter(Mandatory = $true)] [ControllerConfig] $ControllerConfig
+    )
+    $ConfigPath = Get-DefaultCNMPluginsConfigPath
+
+    $Config = @"
+[DRIVER]
+Adapter = $AdapterName
+ControllerIP = $( $ControllerConfig.Address )
+ControllerPort = 8082
+AgentURL = http://127.0.0.1:9091
+VSwitchName = Layered?<adapter>
+
+[AUTH]
+AuthMethod = $( $ControllerConfig.Address )
+
+[KEYSTONE]
+Os_auth_url = $( $OpenStackConfig.AuthUrl() )
+Os_username = $( $OpenStackConfig.Username )
+Os_tenant_name = $( $OpenStackConfig.Project )
+Os_password = $( $OpenStackConfig.Password )
+Os_token =
+"@
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        Set-Content -Path $Using:ConfigPath -Value $Using:Config
+    }
+}
+
+function New-NodeMgrConfig {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session,
+        [Parameter(Mandatory = $true)] [string] $ControllerIP
+    )
+
+    $ConfigPath = "C:\ProgramData\Contrail\etc\contrail\contrail-vrouter-nodemgr.conf"
+    $LogPath = Join-Path (Get-ComputeLogsDir) "contrail-vrouter-nodemgr.log"
+
+    $HostIP = Get-NodeManagementIP -Session $Session
+
+    $Config = @"
+[DEFAULTS]
+log_local = 1
+log_level = SYS_DEBUG
+log_file = $LogPath
+hostip=$HostIP
+
+[COLLECTOR]
+server_list=${ControllerIP}:8086
+
+[SANDESH]
+introspect_ssl_enable=False
+sandesh_ssl_enable=False
+"@
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        Set-Content -Path $Using:ConfigPath -Value $Using:Config
+    }
+}
+
+#Function executed in the remote machine to create Agent's config file.
+function Prepare-AgentConfig {
+    Param (
+        [Parameter(Mandatory = $true)] [string] $ControllerIP,
+        [Parameter(Mandatory = $true)] [string] $VHostIfName,
+        [Parameter(Mandatory = $true)] [string] $VHostIfIndex,
+        [Parameter(Mandatory = $true)] [string] $PhysIfName
+    )
+    $VHostIP = (Get-NetIPAddress -ifIndex $VHostIfIndex -AddressFamily IPv4).IPAddress
+    $PrefixLength = (Get-NetIPAddress -ifIndex $VHostIfIndex -AddressFamily IPv4).PrefixLength
+    $VHostGateway = (Get-NetIPConfiguration -InterfaceIndex $VHostIfIndex).IPv4DefaultGateway
+    $VHostGatewayConfig = if ($VHostGateway) { "gateway=$( $VHostGateway.NextHop )" } else { "" }
```

Comment:
Right now, `if` codeblock is never executed. 

This variable needs to be conditional, because it is connected to single/double interface mode differences.
When using double interface mode, dataplane interface doesn't need to have a L3 gateway, because other Contrail components are in the same subnet.
In single interface mode, there must be an default gateway or otherwise any traffic would be limited to the subnet in which the compute node resides.